src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -209,6 +183,45 @@ private[internal] trait TypeMaps {
           tree1.shallowDuplicate.setType(tpe1)
       }
     }
+
+  }
+
+  abstract class VariancedTypeMap extends TypeMap {
+
+    private[this] var _variance: Variance = Covariant
+
+    def variance_=(x: Variance) = { _variance = x }
+    def variance = _variance
+
+    @inline final def withVariance[T](v: Variance)(body: => T): T = {
+      val saved = variance
+      variance = v
+      try body finally variance = saved
+    }
+    @inline final def flipped[T](body: => T): T = {
+      variance = variance.flip
+      try body
+      finally variance = variance.flip
+    }
+
+    final override def mapOverArgs(args: List[Type], tparams: List[Symbol]): List[Type] =
+      map2Conserve(args, tparams)((arg, tparam) => withVariance(variance * tparam.variance)(this(arg)))
+
+    /** Applies this map to the symbol's info, setting variance = Invariant
+      *  if necessary when the symbol is an alias.
+      */
+    private def applyToSymbolInfo(sym: Symbol, info: Type): Type = {
+      if (!variance.isInvariant && sym.isAliasType)
+        withVariance(Invariant)(this(info))
+      else
+        this(info)
+    }
+
+  }
+
+  object VariancedTypeMap {
+    def unapply(tm: TypeMap): Option[VariancedTypeMap] =
+      if (tm.isInstanceOf[VariancedTypeMap]) Some(tm.asInstanceOf[VariancedTypeMap]) else None
```

Comment:
Why prefer `case VariancedTypeMap(vtm) =>` over `case vtm: VariancedTypeMap =>`? The latter seems fine & simpler to me?

(Side note, you could define this more succinctly as `def unapply(vtm: VariancedTypeMap): Some[VariancedTypeMap] = Some(vtm)`. The pattern matcher will insert the instance-of checks for you before calling the extractor.)

Reply:
Using the `case vtm: VariancedTypeMap` is cleaner, so I have removed this object. This also allows me to revert the change done for the `JavaUniverseForce`.